BatchAuctionOptimization/batchauctions.tex

```diff
@@ -854,80 +819,94 @@ \subsection{Mixed-integer linear programming model I}
 \paragraph{Additional inequalities}
 
 The model does not take into account that there exist dependencies between orders on the same token
-pair, that can heuristically be described as follows:
-Whenever an order with a certain limit price $ \pi^* $ is executed (fully or partially) at some
-determined market price, all orders offering an even better limit price must also be executed.
-Conversely, if some order can not be executed, all orders with worse limit prices may not be
-executed as well.
-This idea induces a natural ordering of the orders that we are now going to formalize:
-let the set of (undirected) token index pairs be denoted by
-$ \itokenpairs := \{(j,k) \in \itokens \times \itokens, j < k\} $, with $ (j,k) \in \itokenpairs $
-representing the pair of tokens $ \tau_j $ and $ \tau_k $.
-
-For every token pair $ (j,k) \in \itokenpairs $, we collect all orders for which $ \tau_j $ is the
-token to be bought and $ \tau_k $ is the token to be sold, \ie all
-$ \omega = (j, k,\, \cdot \,,\, \cdot \,,\, \cdot) \in \orders $,
-and sort them in an increasing order by their limit prices.
-Assuming that there are $ m $ such orders for token pair $ (j,k) $, we get
-$ \pi^{(1)}_{j,k} < \pi^{(2)}_{j,k} < \ldots < \pi^{(m)}_{j,k} $.
-We then pick the binary variables $ z_i $ belonging to the orders under consideration, and permute
-them into the same order, \ie $ (z^{(0)}, z^{(1)}, \ldots, z^{(m)}) $.
-The idea described above consequently translates into precedence constraints of the form
+pair as well as across adjacent token pairs.
+In the following, we will derive valid inequalities that capture some of these dependencies.
+
+First, whenever a sell order with a certain limit price is executed (fully or partially)
+at some determined market price, all orders on the same token pair with a better (\ie lower) limit
+price must also be executed.
+Conversely, if some order can not be executed, all orders with worse (\ie higher) limit prices may
+not be executed as well.
+In particular, let
+$ \omega_{i_1} = (j,k,\cdot,\pi_{i_1}) $ and $ \omega_{i_2} = (j,k,\cdot,\pi_{i_2}) $
+be two orders selling $ \tau_j $ for $ \tau_k $ with $ \pi_{i_1} \le \pi_{i_2} $.
+Then, we can relate the corresponding binary variables as $ z_{i_1} \ge z_{i_2} $.
+
+This idea induces a natural ordering of the orders on every token pair by their limit price.
```

Comment:
Is this a [well-order](https://en.wikipedia.org/wiki/Well-order)?

Reply:
Orders can have the same limit price, so there would be equality for them.